source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -130,6 +130,39 @@ In summary, filters can be used when implementing application flow control that
+| **Advantages**                                                     | - More decoupled implementations.                                                                             | - Full control of the platform behavior.                                                                                           |
+|                                                                    | - Decreased maintenance burden of the platform, maintenance efforts are centered on plugins.                  | - No restrictions from what can be changed.                                                                                        |
+|                                                                    | - Enables reusable and configurable behavior across the platform.                                             | - Suitable for highly customized use cases.                                                                                        |
+|                                                                    | - Easier to maintain long-term due to backward compatibility and versioning policies.                         |                                                                                                                                    |
++--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+
+| **Disadvantages**                                                  | - Can increase complexity in debugging and maintenance because it adds more processing layers.                | - Separation from the core makes long-term maintenance challenging.                                                                |
+|                                                                    | - Limited by existing events and filters available in the Open edX ecosystem.                                 | - High maintenance burden for upgrades and compatibility.                                                                          |
+|                                                                    | - Might increase complexity in plugins for highly customized features.                                        | - Requires deep understanding of the platform to modify it correctly.                                                              |
+|                                                                    | - Might introduce an edx-platform dependency into plugins, mainly when using filters.                         |                                                                                                                                    |
```

Comment:
When implementing new features in plugins, even though you could be using an extension point which should provide enough data about what's being extended, e.g., the enrollment filter provides the `user`, `course_key`, and `enrollment mode`, we might want to perform operations on the data like getting the `course overview` object that inevitably could result in importing edx-platform code into plugins. This is an antipattern, and it's strongly not recommended, but there's no current alternative for doing that kind of operation (this might be solved by learning core in the long term). Is this too specific to add here?

Reply:
How does this sound as a summary / a little bit more detail & clarity?

```suggestion
|                                                                    | - Although it is not recommended to do so, a plugin might introduce an edx-platform dependency, mainly when using filters, when there is no alternative for getting certain types of data.                         |                                                                                                                                    |
```

Reply:
Yes! Thank you. I implemented the suggestion here: https://github.com/openedx/docs.openedx.org/pull/617/commits/d4266da384e73d2dd6713fe1a897503a19c0ab4c